runtime/common/src/main/kotlin/kotlinx/serialization/json/Json.kt

```diff
@@ -61,14 +48,34 @@ private val defaultJsonModule = serializersModuleOf(mapOf<KClass<*>, KSerializer
  * Note that `@ImplicitReflectionSerializer` are used in order to omit `DataHolder.serializer`, but this is a temporary limitation.
  */
 public class Json(
-    @JvmField internal val unquoted: Boolean = false,
-    @JvmField internal val indented: Boolean = false,
-    @JvmField internal val indent: String = "    ",
-    @JvmField internal val strictMode: Boolean = true,
-    val updateMode: UpdateMode = UpdateMode.OVERWRITE,
-    val encodeDefaults: Boolean = true,
+    @JvmField internal val configuration: JsonConfiguration = JsonConfiguration(),
```

Comment:
AFAIR, our main intention for this change was to provide out of the box a 'proper' default for `useArrayPolymorphism=true`. This parameter has a default value, and default value in JsonConfiguration of this flag is also `false`. It is good that users who use argument-less constructor will not experience semantic change, but it is bad that default semantics is not the one that we want. The proposal was to remove default value for this parameter, so users would always explicitly specify their configuration – and at least they will see that they have option of `useArrayPolymorphism` and it is set to `true`. Regarding default instances (plain, unquoted, etc)..., they should have old semantics but probably some warning about unstable/obsolete semantics could be added with the help of deprecated/experimental annotation.